.github/workflows/docker.yml

```diff
@@ -1,90 +1,51 @@
-name: Build Docker Images
+name: Build Notebook IDE images
 
 on:
   pull_request:
+    paths:
+      - ".github/workflows/**"
+      - "notebook-servers/**"
+      - "spark/**"
```

Comment:
IMHO, you shouldn't push an image on a PR. Building, testing and cheking the code style is fine though. 
There could be a case for testing it on the cluster, when you do the PR, but you should have done that before anyways.